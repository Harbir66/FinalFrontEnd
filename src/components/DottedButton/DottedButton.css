.dotted-button{
    width: 19vw;
    margin: 2vh 3vw 0;
    color: #5905CE;
    border: #5905CE 2px dotted;
    height: 6vh;
    border-radius: 1vh;
    font-size: 2vh;
}

.larger{
    width: 50vw;
    background-color: #F4F6FF;
    margin:3vh 0 0 0;
}