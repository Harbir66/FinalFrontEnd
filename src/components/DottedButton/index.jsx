import React from 'react';
import './DottedButton.css';
import PropTypes from 'prop-types';
import Modal from '../Modal';

function DottedButton({ handleNew, text, large }) {
  const [input, setInput] = React.useState('');
  const [isOpen, setIsOpen] = React.useState(false);
  const openModal = () => setIsOpen(true);
  const closeModal = () => setIsOpen(false);
  // console.log(isOpen);
  const handleSubmit = (e) => {
    e.preventDefault();
    handleNew(input);
    setInput('');
    closeModal();
  };
  const className = large ? 'dotted-button larger' : 'dotted-button';
  const modalDiv = large ? (
    <Modal isOpen={isOpen} onClose={closeModal}>
      <h1>Create a new Field</h1>
      <form>
        <label htmlFor="name">
          Name of the new Field
          <input
            type="text"
            id="name"
            value={input}
            onChange={(e) => setInput(e.target.value)}
          />
        </label>
        <button type="button" onClick={closeModal}>
          Cancel
        </button>
        <button onClick={handleSubmit} type="submit">
          Create
        </button>
      </form>
    </Modal>
  ) : (
    <Modal isOpen={isOpen} onClose={closeModal}>
      <h1>Create a new content type</h1>
      <form>
        <label htmlFor="name">
          Name of the content type
          <input
            type="text"
            id="name"
            value={input}
            onChange={(e) => setInput(e.target.value)}
          />
        </label>
        <button type="button" onClick={closeModal}>
          Cancel
        </button>
        <button onClick={handleSubmit} type="submit">
          Create
        </button>
      </form>
    </Modal>
  );
  return (
    <div>
      <button onClick={openModal} type="button" className={className}>
        {text}
      </button>
      {modalDiv}
    </div>
  );
}

export default DottedButton;

DottedButton.propTypes = {
  text: PropTypes.string.isRequired,
  large: PropTypes.bool.isRequired,
  handleNew: PropTypes.func.isRequired,
};
