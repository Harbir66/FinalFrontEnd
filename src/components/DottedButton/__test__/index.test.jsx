import React from 'react';
import { render } from '@testing-library/react';
import DottedButton from '..';

describe('DottedButton', () => {
  it('should render correctly with large attribute ', () => {
    const { asFragment } = render(
      <DottedButton handleNew={jest.fn()} text="test" large />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render correctly without large attribute ', () => {
    const { asFragment } = render(
      <DottedButton handleNew={jest.fn()} text="test" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
