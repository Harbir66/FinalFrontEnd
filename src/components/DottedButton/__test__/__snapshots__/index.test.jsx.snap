// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DottedButton should render correctly with large attribute  1`] = `
<DocumentFragment>
  <div>
    <button
      class="dotted-button larger"
      type="button"
    >
      test
    </button>
  </div>
</DocumentFragment>
`;

exports[`DottedButton should render correctly without large attribute  1`] = `
<DocumentFragment>
  <div>
    <button
      class="dotted-button"
      type="button"
    >
      test
    </button>
  </div>
</DocumentFragment>
`;
